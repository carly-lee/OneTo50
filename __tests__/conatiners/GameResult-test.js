import 'react-native';
import React from 'react';
import GameResult from 'containers/gameresult';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe( 'GameResult', ()=>{
  const onRestart = jest.fn();

  it( 'renders correctly',()=>{
    const tree = renderer.create( <GameResult finalTime={'34.50'} onRestart={onRestart} /> );
    expect( tree ).toMatchSnapshot();
  });

  it( 'displays the result from props', ()=>{
    const wrapper = shallow( <GameResult finalTime={'37.48'} onRestart={onRestart} /> );
    expect( wrapper.find('Text').length ).toEqual( 1 );
    expect( wrapper.find('Text').children().node ).toEqual( '37.48' );
  });

  it( 'calls onRestart() when a user presses the button', ()=>{
    const wrapper = shallow( <GameResult finalTime={1.50} onRestart={onRestart} /> );
    wrapper.find( 'Button' ).simulate( 'press' );
    expect( onRestart ).toBeCalled();
  });
});