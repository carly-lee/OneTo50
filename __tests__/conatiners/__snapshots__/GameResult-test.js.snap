exports[`GameResult renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 15,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 48,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }>
    34.50
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="Restart"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }>
        Restart
      </Text>
    </View>
  </View>
</View>
`;
